### PR TITLE
Cleanup more cli leftovers

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #33748

Remove more of the `docker/cli` cross build dependencies, and other leftovers from removing `docker/cli`.

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]
LGTM if everything is green.

File: Dockerfile [hunk 1]
```diff
@@ -238,7 +214,7 @@ RUN ./contrib/download-frozen-image-v2.sh /docker-frozen-images \
 # Please edit hack/dockerfile/install-binaries.sh to update them.
 COPY hack/dockerfile/binaries-commits /tmp/binaries-commits
 COPY hack/dockerfile/install-binaries.sh /tmp/install-binaries.sh
-RUN /tmp/install-binaries.sh tomlv vndr runc containerd tini proxy bindata dockercli
```
Comment (tiborvass) [inline Dockerfile]: isn't bindata used for dockerd on windows?

Comment (dnephin) [inline Dockerfile]: Nope, it was added for the Compose file jsonschema. The windows embedding uses some other `windres` tool to embed things.

**Comment by dnephin** [comment]
docker-py test is failing across a few branches, so unrelated to this change

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

**Review by boaz0 — APPROVED** [review]
👍

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**